documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
-- `ListByParent()` - where "Parent" is a context specific suffix. It lists all resource under a parent.
+- [ ] **M1011**: Resource Model name *MUST* be same as singular form of the resource type. Like for virtualMachines resource type it should be VirtualMachine.
 
-&#x25a2; *M1003*: Terminology in Swagger and Azure Portal MUST be the same.
+- [ ] **M1012**: Parameter names in the apis *MUST* be name of the resource like virtualMachines\{virtualMachine} or virtualMachines\{virtualMachineName} both are good example but virtualMachines\{vm} is bad example. 
 
+- [ ] **M1013**: Any action name *MUST* be same as what is defined in the Rest APIs, like regenerateAdminKeys should be regenerateAdminKeys in the Swagger as well, where as regenerateKeys is bad example.
 
 
 ## Operation Semantics - Swagger Checklist ##
 
-&#x25a2; *M2000*: A Swagger specification MUST be accompanied with a document that describes management scenarios that can be realized using operations described in the Swagger.
+### MUST
 
-&#x25a2; *M2001*: Read only parameters MUST be labeled as read only in Swagger.
+- [ ] **M2000**: TODO: A Swagger specification *MUST* be accompanied with a document that describes management scenarios (what does this mean? x-ms-examples or a multi operation scenario?) that can be realized using operations described in the Swagger.
 
-&#x25a2; *M2002*: Boolean parameters MUST be represented as boolean in Swagger. Boolean parameters MUST NOT be represented as strings. Similarly, this requirements applies to other primitive data types such as integer, float and double. These primitive data types MUST be represented using [Swagger primitive data types](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types).
+- [ ] **M2001**: Read only parameters and model properties *MUST* be labeled as `"readOnly": true` in Swagger.
 
-&#x25a2; *M2003*: Long running operations MUST be modeled as [long running operations](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#longrunning) in Swagger.
+- [ ] **M2002**: Boolean parameters *MUST* be represented as boolean in Swagger. Boolean parameters *MUST* NOT be represented as strings. Similarly, this requirements applies to other primitive data types such as integer, float and double. These primitive data types *MUST* be represented using [Swagger primitive data types](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types).
 
-&#x25a2; *M2004*: Required parameters MUST be labeled as required in Swagger.
+- [ ] **M2003**: If a format is specified along with a type, it *MUST* be from one of the supported formats - "int64", "int32", "float", "decimal", "double", "byte", "base64url", "unixtime", "date", "date-time", "duration", "date-time-rfc1123", "uuid", otherwise in codegeneration the artifact will be of the specified type and not as per the desired format. 
```

Comment:
The **must** here should be that any data returned that can be accurately described by one of the supported formats and/or types, it must be so. 

On the other hand, we must not prevent a swagger document from being more specific in the format specified than what we happen to support at that given point in time (which the current description does).